### ISSUE TITLE
fedora 40 -> fedora 41?

### ISSUE THREAD

**Issue by arrl-mtk** [WhitewaterFoundry/Fedora-Remix-for-WSL#227]
how are major version upgrades handled?  does the "dnf system-upgrade" process work in this special environment?  do we need to make a new purchase?  is there some other upgrade process?

**Comment by crramirez** [comment]
Yes, the dnf system-upgrade process works, but with some differences, please check the instructions here: https://www.whitewaterfoundry.com/blog/2024/04/23/fedora-remix-for-wsl-40-released

We will update these instructions for the 40 -> 41 migration soon. Also we publish a new updated version on the store, replacing the current one.

The license is permanent, you have the right to receive the updates.

**Comment by shoffmeister** [comment]
FWIW, upgrading from 40 to 41 removed the RPM version lock on Mesa. This meant that the special build of Mesa provided by this remix of Fedora was "removed" and pushed towards the latest 41 Fedora-native Mesa.

As Fedora 41 itself does not ship with the right Mesa drivers (d3d12), WSLg (graphics) performance after "dnf system-upgrade" is atrocious.

If you want to retain good graphics performance, your options are
* wait for a proper 41 release of this (excellent) Fedora remix (which I expect to include the d3d12 drivers, as was the case for previous releases)
* "dnf system-upgrade" and either of
  * install some COPR mesa
  * build and install your own Mesa from scratch
  * accept atrocious graphics performance

**Comment by crramirez** [comment]
I also recommend you to wait that we release the mesa drivers with d3d12. This process is usually what delay us to ship a new image, but this time it is the dnf 5.

Probably we will ship first the upgraded mesa drivers so you can upgrade and then the image with Fedora 41.

**Comment by crramirez** [comment]
The updated mesa version 24.2.5-1_wsl_2 is ready to download. Just run the `update.sh` script to get them.

```
name of display: :0
TU: error: ../src/freedreno/vulkan/tu_knl.cc:315: failed to open device /dev/dri/renderD128 (VK_ERROR_INCOMPATIBLE_DRIVER)
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Microsoft Corporation (0xffffffff)
    Device: D3D12 (NVIDIA GeForce GTX 970) (0xffffffff)
    Version: 24.2.5
    Accelerated: yes
    Video memory: 36757MB
    Unified memory: no
    Preferred profile: core (0x1)
    Max core profile version: 4.6
    Max compat profile version: 4.6
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.1
OpenGL vendor string: Microsoft Corporation
OpenGL renderer string: D3D12 (NVIDIA GeForce GTX 970)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 24.2.5
OpenGL core profile shading language version string: 4.60
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.6 (Compatibility Profile) Mesa 24.2.5
OpenGL shading language version string: 4.60
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile

OpenGL ES profile version string: OpenGL ES 3.1 Mesa 24.2.5
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.10
```
The error in the first line is expected

**Comment by shoffmeister** [comment]
... and the updated Mesa drivers work well :)

Many thanks!

**Comment by arrl-mtk** [comment]
i'm confused.  the msoft store says version 41.0.0 has been 'Upgraded to Fedora 41'.  and that i can update it by running /usr/local/bin/update.sh.  i've run that under both my ID and under root.  it didn't report any problems.  but /etc/redhat-release still says v40.  the installed kernel is v5.15 but the kernel on the half dozen other fedora instances i manage is 6.11.

**Comment by crramirez** [comment]
Hello @arrl-mtk ,

Sorry for the confusion. The text in the store should be more detailed. What update.sh does is to update the customization scripts tailored to WSL. For example it can upgrade the mesa libraries to the latest that we have in the repo.

To upgrade from 40 to 41 you must follow this instructions : https://www.whitewaterfoundry.com/blog/2024/04/23/fedora-remix-for-wsl-40-released

The kernel is managed by Microsoft with the WSL releases. You can follow them here: https://github.com/microsoft/WSL/releases/tag/2.3.24

The kernel version 6 will be available any time soon. But each time that Microsoft releases it, a bunch of things get broken.

**Comment by crramirez** [comment]
Yes I recommended to wait because the mesa drivers, to avoid you the work of compiling them yourself.

The store now has Fedora 41 but without a manual upgrade, you can access it if you unregister your distro losing everything that you have there and create it again. In this case it will be Fedora 41

**Comment by arrl-mtk** [comment]
not sure what you mean by 'manual upgrade'.  as a longtime fedora user (since version 5 and i even started a job a year ago that had a version 3 system in production!), i'm lazy :-).  i want either directly or the equivalent of "dnf system-upgrade reboot".  in general, how do you envision major upgrade versions without disturbing the current install?

**Comment by crramirez** [comment]
By manual upgrade I mean following the steps that are in the link that aren't so different to the upgrade process in a Fedora installation in bare metal. Before doing a `dnf system-upgrade reboot` you need to take some steps. And in the case of WSL you need more because there is no reboot. I've seen users that upgrade Fedora with systemd enabled and do the system-upgrade reboot, and it works. But not always work. The method in the blog post is a method that we have tested version after version with a very high success rate.

WSL is a different environment, they are basically live containers and they should be treated that way if you want to minimize data loss.

**Comment by f1amy** [comment]
When will official update instructions for 40->41 be published? Sorry if it was already said elsewhere

**Comment by arrl-mtk** [comment]
I followed the 39->40 upgrade instructions for 40->41 and it was both easy and quick.  /mark

**Comment by crramirez** [comment]
@f1amy

I apologize for the unusual delay. We hope have it the next week. Meanwhile, as others have pointed you can use the previous one changing the versions. It's exactly the same procedure.

Regards,
Carlos

**Comment by crramirez** [comment]
Hello @f1amy,

Here are the official instructions: https://www.whitewaterfoundry.com/blog/2024/10/29/fedora-remix-for-wsl-41-released

Best Regards,
Carlos

**Comment by f1amy** [comment]
Thanks!
